src/main/groovy/com/twcable/grabbit/jcr/JcrNodeDecorator.groovy

```diff
@@ -136,6 +136,37 @@ class JcrNodeDecorator {
     }
 
     /**
+     * Finds out nearest versionable ancestor for a node and performs a checkout
+    */
+    void checkoutNode() {
+        try {
+            JcrNodeDecorator decoratedVersionableAncestor = findVersionableAncestor()
+            if (decoratedVersionableAncestor && !decoratedVersionableAncestor.isCheckedOut()) {
+                decoratedVersionableAncestor.session.workspace.versionManager.checkout(decoratedVersionableAncestor.path)
+            }
+        }
+        catch (Exception exception) {
```

Comment:
So luckily .. first 2 extend `RepositoryException` .. so you'd just need to do catch `RepositoryException` .. in-fact IMO if we get any exception that is other than a type of `RepositoryException` ... we should break.